qt/scientific_interfaces/General/StepScan.cpp

```diff
@@ -599,6 +599,7 @@ void StepScan::generateCurve(const QString &var) {
     MatrixWorkspace_sptr top = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(m_plotWSName);
     MatrixWorkspace_sptr bottom = norm->getProperty("OutputWorkspace");
     top /= bottom;
+    AnalysisDataService::Instance().addOrReplace(m_plotWSName, top);
```

Comment:
I've thought about whether adding an explicit call to divide is worth it or not. Sadly the OutputWorkspace name from norm (`bottom->getName()`) returns an empty string and the bottom workspace is not in the ADS. So I think I'm happy with this addOrReplace call as it's in a relatively niche part of the codebase.

Reply:
Can I ask how this came about as a fix for a cppcheck warning?

Reply:
Cppcheck was complaining that top is an `unreadVariable`. I did something a bit cheaty where I add the top workspace into the ADS (even though it already exists in the ADS), just to remove the unreadVariable warning. I'm happy to remove this and put back in the cppcheck suppression as its probably a false positive